ui/server.py

```diff
@@ -61,8 +67,69 @@ def _get_rules():
     return rulesdict, cat
 
 
+# Store JWT's value for every connected user
+registered_tokens = []
+
+
+@app.before_request
+def before_request():
+    """
+        Treat all incoming requests before-hand. 
+        If the user is not yet logged in, he/she will be redirected towards the login page.
+    """
+    if(HTTPS):
+        token = request.cookies.get('AUTH')
+        if(token not in registered_tokens):
+            if(request.endpoint != 'login' and '/res/' not in request.path):
+                return render_template('login.html',
+                                       msg='🔒 Enter your secret key to access the scanner:')
+
 # ################### ROUTES ####################
 
+
+@app.route('/login', methods=['POST', 'GET'])
+def login():
+    # If the HTTPS protocol is not in use, then the login feature will be disabled.
+    if(not HTTPS):
+        redirect(url_for('root'))
+    else:
+        if(request.method == 'POST'):
+            auth_key = request.form['auth_key']
+            if(auth_key != os.getenv('AUTH_KEY')):
+                redirect(url_for('login'))
+                return render_template('login.html',
+                                       msg='❌ Wrong key, please try again:')
+            # We generate a UUID to be saved as a JWT's value
+            import uuid
```

Comment:
why do you import it here? I think you can put it with the others at the first lines of the file